### PR TITLE
[Intake] require `message` or `type` for error.exception

### PR DESCRIPTION
**Pull request by simitt**
Make either `message` or `type` required for `error.exception`,
without changing the logic for grouping_key.

implements #1297

### CODE REVIEW

**Comment by roncohen** [comment]
@elastic/apm-ui heads up, this affects you.

File: model/error/event_test.go [hunk 1]
```diff
@@ -188,17 +189,18 @@ func TestErrorEventDecode(t *testing.T) {
 			},
 		},
 	} {
-		for _, decodeFct := range []func(interface{}, error) (transform.Transformable, error){V1DecodeEvent, V2DecodeEvent} {
+		for idx, decodeFct := range []func(interface{}, error) (transform.Transformable, error){V1DecodeEvent, V2DecodeEvent} {
 			transformable, err := decodeFct(test.input, test.inpErr)
 
 			if test.e != nil {
+				fmt.Println(idx)
```
Comment (axw) [inline model/error/event_test.go]: did you mean to leave that Println in?

Comment (simitt) [reply model/error/event_test.go]: Thanks for catching this, will remove in a follow up.

**Comment by roncohen** [comment]
Sorry, it took so long to review this @simitt.

I just realized we now allow `"exception": {"message": null}` without `type` being specified, e.g.:

```
{ "error": {"id": "cdefab0123456789", "transaction_id": "0000555544446666", "timestamp": "2018-08-09T14:59:05.999Z", "exception": {"message": null}}}
```

Allowing `null` for optional values is in line with how we do optional values, however, we probably shouldn't allow `"message": null` if `type` isn't set. 

I did a quick experiment, and it looks like we can get to what we want by adding something like:
```
"anyOf": [
  {"required": ["message"], "properties": {"message": {"type": "string", "minLength": 1}}},
  {"required": ["type"], "properties": {"type": {"type": "string", "minLength": 1}}}
]
```

**Pull request merged**